api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -330,6 +330,10 @@ message Bootstrap {
   // Optional path to a file with performance tracing data created by "Perfetto" SDK in binary
   // ProtoBuf format. The default value is "envoy.pftrace".
   string perf_tracing_file_path = 33;
+
+  // Optional overriding of default regex engine. The value must be the name of one of the
+  // regex engine factories initialized through a bootstrap extension.
+  string default_regex_engine = 34;
```

Comment:
Is there a way to override regex engine other than default in each regex matcher? We don't use name to identify extensions anymore, so shouldn't this just be an `TypedExtensionConfig`?

Reply:
> Is there a way to override regex engine other than default in each regex matcher?

The bootstrap option provides the way configuring regex engine in one single place other than write them in each `RegexMatcher`.

> We don't use name to identify extensions anymore, so shouldn't this just be an TypedExtensionConfig?

Ok, I may take some time to change it from `string` to `TypedExtensionConfig`.

Reply:
Done.

Reply:
> The bootstrap option provides the way configuring regex engine in one single place other than write them in each `RegexMatcher`.

My question is, how to override in each `RegexMatcher`? It doesn't seems to have such a way in this PR. Just not over complicate this that's fine though.

Reply:
Yes, the patch does not provide the way to override in each `RegexMatcher`. The original `GoogleRE2` message has deprecated all its fields, so I think we do not need overriding methods currently.